### DISCUSSION TITLE
How to upgrade on a web host?

### DISCUSSION THREAD

**Discussion by Pyromane** [Part-DB/Part-DB-server#176] [Q&A]
I have an existing Part-DB 0.5.12 installation on a web host without SSH access and would like to migrate to the Part-DB symfony version. What is the correct process here?

**Comment by jbtronics** [comment]
The problem is that every modern web framework (and the web apps using them) are assuming that you have SSH access to your server. Running complex web apps on shared web hosters, where you do not have SSH access to, is pretty uncommon today (as even if you use something like Plesk or ISPconfig for managing your shared server, you can setup restricted users, which can be used via SSH).

Installing/Updating this Part-DB without SSH access is not impossible but not really recommended, as I guess it is pretty complicated:
As you can not install the needed dependencies and build autoloader and frontend files on the server, you have to generate them somewhere else and copy them to the webserver manually. You could do this by setting up a Part-DB instance locally or extract the needed files from the docker images (which contains freshly build versions of the needed files).
I think the only folders which are need to be copied here are the `vendor/` folder and the `public/build/`.

With that files the software itself should work. The problem is that you have to run the DB migrations (which without Part-DB can not be used, or will just throw errros), which is as far as I know only possible from the command line. In principle you could maybe use one of the PHP based web consoles (like [this](https://github.com/antonmedv/console)) to run this via a web browser (but many shared webhosters block `proc_open` command, so this might not be usable).

In general it is maybe easier to change to another web hoster who allows SSH access, if possible (especially as you have to do all this work again every time you want to update Part-DB)
